File: src/main/java/com/pretty/platform/catalog/domain/model/events/ProductCreatedEvent.java
package com.pretty.platform.catalog.domain.model.events;

import java.time.LocalDateTime;

/**
 * Event raised when a product is created
 */
public class ProductCreatedEvent {
    private final Long productId;
    private final LocalDateTime occurredOn;

    public ProductCreatedEvent(Long productId) {
        this.productId = productId;
        this.occurredOn = LocalDateTime.now();
    }

    public Long getProductId() { return productId; }
    public LocalDateTime getOccurredOn() { return occurredOn; }
}

File: internal/products/domain/model/events/product_created_event.go
package events

import (
	"go-service/internal/products/domain/model/valueobjects"
	"time"
)

type ProductCreatedEvent struct {
	productID  valueobjects.ProductID
	occurredOn time.Time
}

func NewProductCreatedEvent(productID valueobjects.ProductID) ProductCreatedEvent {
	return ProductCreatedEvent{
		productID:  productID,
		occurredOn: time.Now(),
	}
}

func (e ProductCreatedEvent) ProductID() valueobjects.ProductID { return e.productID }
func (e ProductCreatedEvent) OccurredOn() time.Time             { return e.occurredOn }
